src/main/java/com/conveyal/r5/analyst/scenario/ModificationTypeResolver.java

```diff
@@ -0,0 +1,75 @@
+package com.conveyal.r5.analyst.scenario;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.jsontype.impl.TypeIdResolverBase;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+
+import java.io.IOException;
+
+/**
+ * This establishes a mapping between type codes embedded in JSON objects and Java Modification types on R5 workers.
+ * It is used to achieve polymorphism in serialization and deserialization to and from JSON.
+ * Almost everything here can be achieved with Jackson annotations @JsonTypeInfo and @JsonTypes on the classes.
+ * But there's one exception: the custom modification type that pipes arbitrary JSON through the backend to the R5
+ * worker. In that case we need to overwrite the type code "custom" with the type code we want the worker to see, which
+ * might be completely unknown to the backend.
+ *
+ * Actually we might be able to perform the bulk of the class/type string mappings using those annotations, and just
+ * override the behavior only on the CustomModification (with a TypeIdResolver that only handles the CustomModification
+ * case). That could be done in addition to moving the R5 CustomModification to the backend, so it's clearly outside R5.
+ *
+ * Created by abyrd on 2019-03-15
+ */
+public class ModificationTypeResolver extends TypeIdResolverBase {
+
+    /**
+     * A mapping between all Modification types known by this worker and their JSON type codes.
+     * This table is not really needed since there's a deterministic mapping between CamelCase and kebab-case here.
+     * We could just perform that mapping on the fly in methods, but I suppose this is more clear and efficient.
+     */
+    private static final BiMap<String, Class<? extends Modification>> modificationTypes = new ImmutableBiMap.Builder()
+            .put("adjust-speed", AdjustSpeed.class)
+            .put("adjust-frequency", AdjustFrequency.class)
+            .put("adjust-dwell-time", AdjustDwellTime.class)
+            .put("add-trips", AddTrips.class)
+            .put("remove-trips", RemoveTrips.class)
+            .put("reroute", Reroute.class)
+            .put("remove-stops", RemoveStops.class)
+            .put("road-congestion", RoadCongestion.class)
+            .put("set-fare-calculator", SetFareCalculator.class)
```

Comment:
Fare Calculator is now set in the profile request, not as a modification. So we can probably delete this annotation and the Class below.